File: 299-bulls-and-cows/leochang/299.cpp
class Solution {
public:
	string getHint(string secret, string guess) {
		if (secret.empty()) return "0A0B";
		int A = 0, B = 0;
		vector<bool> isA(guess.size(), false);
		vector<bool> isMatch(secret.size(), false);

		for(int i = 0; i < guess.size(); i++) {
			if (guess[i] == secret[i]) {
				A++;
				isMatch[i] = true;
				isA[i] = true;
				continue;
			}
		}
		for(int i = 0; i < guess.size(); i++) {
			if (isA[i]) continue;
			for (int j = 0; j < secret.size(); j++) {
				if (guess[i] == secret[j] && !isMatch[j]) {
					isMatch[j] = true;
					B++;
					break;
				}
			}
		}

		stringstream ss;
		ss << A << "A" << B << "B";
		return ss.str();
	}
};

File: 299-bulls-and-cows/ken/299.py
class Solution(object):
    def getHint(self, secret, guess):
        """
        :type secret: str
        :type guess: str
        :rtype: str
        """

        A = 0
        B = 0
        letterCnt = {}

        for i in xrange(len(secret)):
            if secret[i] == guess[i]:
                A += 1
            else:
                if secret[i] not in letterCnt:
                    letterCnt[secret[i]] = 0
                elif letterCnt[secret[i]] < 0:
                    B += 1
                letterCnt[secret[i]] += 1

                if guess[i] not in letterCnt:
                    letterCnt[guess[i]] = 0
                elif letterCnt[guess[i]] > 0:
                    B += 1
                letterCnt[guess[i]] -= 1

        return "%dA%dB" % (A, B)
